salt/modules/iptables.py

```diff
@@ -220,6 +220,17 @@ def maybe_add_negation(arg):
                 rule.append('--name {0}'.format(kwargs['name']))
         del kwargs['match']
 
+    if 'match-set' in kwargs:
+        if isinstance(kwargs['match-set'], basestring):
```

Comment:
Please change this to `if isinstance(kwargs['match-set'], six.string_types):` and add `from salt.ext import six` at the top.